python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -701,6 +703,58 @@ def cummax(self):
         """Get the column-wise cumulative maximum value in each group."""
         return self.agg("cummax")
 
+    def shift(self, periods=1, freq=None, axis=0, fill_value=None):
+        """
+        Shift each group by ``periods`` positions.
+
+        Parameters
+        ----------
+        periods : int, default 1
+            Number of periods to shift.
+        freq : str, unsupported
+        axis : 0, axis to shift
+            Shift direction. Only row-wise shift is supported
+        fill_value : scalar or list of scalars, optional
+            The scalar value to use for newly introduced missing values. Should
+            match the dtype of columns to fill.
+
+        Returns
+        -------
+        Series or DataFrame
+            Object shifted within each group.
+
+        Notes
+        -----
+        Parameter ``freq`` is unsupported.
+
+        Parameter ``fill_value`` can be specified with ``None``, a single value
+        or multiple values. When multiple values are specified, the number of
+        values should match the number of columns to shift. Each column shifted
+        is filled with the corresponding value in the value list.
```

Comment:
It would be good to include this in the Parameters section since it's important for someone passing a non-scalar value.

Reply:
Will move to Parameter section.

We actually can pass in a `cudf.Scalar` as long as `scalar.dtype == column.dtype`. The only thing we don't support is casting `scalar` dtype.